src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java

```diff
@@ -44,19 +48,42 @@ public void check(JavaClass javaClass, ConditionEvents events) {
     }
   }
 
+  @SneakyThrows
+  private Set<Class<?>> extractAllClassesReferencedFromPluginXmlAttributes() {
+    val classLoader = Thread.currentThread().getContextClassLoader();
+    val db = DocumentBuilderFactory.newInstance().newDocumentBuilder();;
+    val doc = db.parse(classLoader.getResourceAsStream("META-INF/plugin.xml"));
+    val nodeList = doc.getElementsByTagName("*");
+    Set<Class<?>> result = new HashSet<>();
+    for (int i = 0; i < nodeList.getLength(); i++) {
+      val node = nodeList.item(i);
+      val attributes = node.getAttributes();
+      for (int j = 0; j < attributes.getLength(); j++) {
+        val attribute = attributes.item(j);
+        val maybeFqcn = attribute.getNodeValue();
+        try {
+          val clazz = Class.forName(maybeFqcn, /* initialize */ false, classLoader);
+          result.add(clazz);
+        } catch (ClassNotFoundException e) {
+          /* ignore */
+        }
+      }
+    }
+    return result;
+  }
+
   @Test
   public void all_classes_should_be_referenced() {
+    val classesReferencedFromPluginXmlAttributes = extractAllClassesReferencedFromPluginXmlAttributes().toArray(Class[]::new);
     classes()
         .that().resideOutsideOfPackages(
             // Classes in *.impl.* packages may be instantiated via RuntimeBinding
             "..impl..",
-            // For some reason, ArchUnit doesn't see accesses to static fields of classes from frontend defs package
-            "com.virtuslab.gitmachete.frontend.defs",
-            // Classes in frontend:file may be referenced from plugin.xml
-            "com.virtuslab.gitmachete.frontend.file..")
-        // Actions may be referenced from plugin.xml
-        .and().haveSimpleNameNotEndingWith("Action")
-        // For some reason, references to SlideInDialogKt aren't detected (Kotlin class?)
+            // For some reason, ArchUnit (com.tngtech.archunit.core.domain.JavaClass.getAccessesFromSelf)
+            // doesn't see accesses to static fields
+            "com.virtuslab.gitmachete.frontend.defs")
+        .and().doNotBelongToAnyOf(classesReferencedFromPluginXmlAttributes)
+        // For some reason, ArchUnit doesn't see access to the static field GIT_REF_PROTOTYPE_VALUE
```

Comment:
hmmm. y dont we bring it into the class? probably into the companion object. would it solve that problem?

If this one remains pls use some {@link classreference}  or smth (if possible)

Reply:
Inlined the constant completely